waku/v2/protocol/filter/client.go

```diff
@@ -248,11 +286,24 @@ func (wf *WakuFilterLightNode) Subscribe(ctx context.Context, contentFilter Cont
 		return nil, ErrNoPeersAvailable
 	}
 
-	err := wf.request(ctx, params, pb.FilterSubscribeRequest_SUBSCRIBE, contentFilter)
+	pubSubTopicMap, err := contentFilterToPubSubTopicMap(contentFilter)
 	if err != nil {
 		return nil, err
 	}
-	return wf.subscriptions.NewSubscription(params.selectedPeer, contentFilter.Topic, contentFilter.ContentTopics), nil
+	subscriptions := make([]*SubscriptionDetails, 0)
+	for pubSubTopic, cTopics := range pubSubTopicMap {
+		var cFilter ContentFilter
+		cFilter.Topic = pubSubTopic
+		cFilter.ContentTopics = cTopics
+		//TO OPTIMIZE: Should we parallelize these, if so till how many batches?
+		err := wf.request(ctx, params, pb.FilterSubscribeRequest_SUBSCRIBE, cFilter)
```

Comment:
What happens if one of the requests in this loop fail and previous iterations were succesful?

Reply:
Good point, now that i am relooking at it...maybe we can collect the errors and return them to user indicating which contentTopics failed?